includes/User.php

```diff
@@ -592,6 +592,40 @@ public static function whoIs( $id ) {
 	}
 
 	/**
+	 *
+	 * @param $ids Array User IDs
+	 * @return Array User ID to User name mapping
+	 */
+	public static function whoAre( Array $ids, $source = DB_SLAVE ): Array {
+		global $wgExternalSharedDB;
+		$sdb = wfGetDB( $source, [], $wgExternalSharedDB );
+		$res = $sdb->select(
+			'`user`',
+			[ 'DISTINCT user_id', 'user_name' ],
+			[ 'user_id' => $ids ],
+			__METHOD__
+		);
```

Comment:
Should we cache the results in some way?

Reply:
I though about it.  We can order and serialize the input array, `sha1` it, include in the memcache key, and cache the result for some short period of time.  Caching for long period of time will result with a large number of keys and pose a few questions such as:

* how to selectively invalidate cache after a user rename?
* what is the chance that we will query for the exact same set of names more than once?